osenv.go

```diff
@@ -67,16 +67,34 @@ var testingVariables = []string{
 }
 
 func (s *OsEnvSuite) setEnviron() {
-	var envList []string
+	var isWhitelisted func (string) bool
 	switch runtime.GOOS {
 	case "windows":
-		envList = windowsVariables
+		// Lowercase variable names for comparison as they are case
+		// insenstive on windows. Fancy folding not required for ascii.
+		lowerEnv := make(map[string]string)
```

Comment:
this should probably be map[string]struct{} ... I'm sure the size isn't a big deal, but since we never use the value, it would be nice to make it a little more clear that the value doesn't matter.